miniapps/spde/visualizer.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2010-2022, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details
+
+#include "visualizer.hpp"
+
+namespace mfem {
+namespace spde {
+
+void Visualizer::ExportToParaView() {
+  ParaViewDataCollection paraview_dc("SurrogateMaterial", mesh_);
+  paraview_dc.SetPrefixPath("ParaView");
+  paraview_dc.SetLevelsOfDetail(order_);
+  paraview_dc.SetCycle(0);
+  paraview_dc.SetDataFormat(VTKFormat::BINARY);
+  paraview_dc.SetHighOrderOutput(true);
+  paraview_dc.SetTime(0.0);  // set the time
+  paraview_dc.RegisterField("random_field", &g1_);
+  if (is_3D_) {
+    paraview_dc.RegisterField("topological_support", &g2_);
+    paraview_dc.RegisterField("imperfect_topology", &g3_);
+    paraview_dc.RegisterField("level_set", &g4_);
+  }
+  paraview_dc.Save();
+}
+
+void Visualizer::SendToGLVis() {
```

Comment:
I would suggest using the `window_geometry` directive (see examples 31 and 32 for reference) for GLVis output to position the windows so that they don't appear on top of each other.